experimental/packages/opentelemetry-instrumentation-http/test/functionals/utils.test.ts

```diff
@@ -458,11 +464,37 @@ describe('Utility', () => {
         {
           component: 'http',
           semconvStability: SemconvStability.OLD,
+          enableSyntheticSourceDetection: false,
         },
         diag
       );
       assert.strictEqual(attributes[SEMATTRS_HTTP_TARGET], '/user/?q=val');
```

Comment:
It would be nice to assert that `attributes[ATTR_USER_AGENT_SYNTHETIC_TYPE]` is *not* set in this case.